dbt/adapters/databricks/impl.py

```diff
@@ -256,47 +255,63 @@ def get_relations_without_caching(self, relation: DatabricksRelation) -> Table:
 
         # if there are any table types to be resolved
         if any(not row[3] for row in new_rows):
-            # Get view names and create a dictionay of view name to materialization
+            # Get view names and tables and create a dictionary of name to materialization
+            relation_all_tables = self.Relation.create(
+                database=relation.database, schema=relation.schema, identifier="*"
+            )
             with self._catalog(relation.database):
                 views = self.execute_macro(SHOW_VIEWS_MACRO_NAME, kwargs=kwargs)
+                tables = self.execute_macro(
+                    SHOW_TABLE_EXTENDED_MACRO_NAME, kwargs={"relation": relation_all_tables}
+                )
 
             view_names: Dict[str, bool] = {
                 view["viewName"]: view.get("isMaterialized", False) for view in views
             }
 
+            def parse_type(information: str) -> str:
+                type_entry = [
+                    entry.strip()
+                    for entry in information.split("\n")
+                    if entry.split(":")[0] == "Type"
+                ]
+                return type_entry[0] if type_entry else ""
+
+            table_names: Dict[str, bool] = {
+                table["tableName"]: (parse_type(table["information"]) == "STREAMING_TABLE")
+                for table in tables
+            }
+
             # a function to resolve an unknown table type
-            def typeFromNames(
-                database: Optional[str], schema: str, name: str
-            ) -> DatabricksRelationType:
-                if name in view_names:
+            def typeFromNames(database: Optional[str], name: str) -> DatabricksRelationType:
+                if is_hive_metastore(database):
```

Comment:
this isn't a safe reordering.  We can only default to table if we know its hive metastore, and not a view.

Reply:
Ah, ok.